### ISSUE TITLE
glsl300 support

### ISSUE THREAD

**Issue by zouyaoji** [RaymanNg/3D-Wind-Field#71]
Please do NOT post a issue asking for complete code specially for your purpose, this kind of issue is not welcomed.

**Describe your problem**
The latest cesium has been upgraded to webgl2, and the glsl syntax is invalid. I tried to upgrade, but it failed. Do you have time to give me some advice?

**Steps to reproduce**
What did you do before your problem occurred?

https://github.com/zouyaoji/3D-Wind-Field/tree/master/Cesium-3D-Wind/glsl

**Comment by 3DGISKing** [comment]
experienced the same

**Comment by 3DGISKing** [comment]
had to understand why this warnings happens

![image](https://github.com/RaymanNg/3D-Wind-Field/assets/26606704/4a41a3bc-3e45-4c0d-b73e-56e78c04c53a)

fixed by replacing LUMINANCE with RED
![image](https://github.com/RaymanNg/3D-Wind-Field/assets/26606704/39e4abe4-093b-47e1-ace6-8017ac02c2e9)

**Comment by zouyaoji** [comment]
@3DGISKing 
Works for me too, thanks a lot

Is there anything specific about LUMINANCE and RED?

**Comment by zouyaoji** [comment]
The reason for changing the texture format from `Cesium.PixelFormat.LUMINANCE` to `Cesium.PixelFormat.RED` is that you were using single-channel texture data in your shader, and `Cesium.PixelFormat.LUMINANCE` represents a luminance texture, which has only one channel. In your shader, you were using the `texture2D` function to sample the texture data, which requires at least two-channel texture data to work properly.

By changing the texture format to `Cesium.PixelFormat.RED`, you specified a single-channel texture format that matches the texture sampling function used in your shader. This avoids errors in the shader.

In summary, by changing the texture format to match the texture sampling function used in the shader, you resolved the warning and achieved the desired effect.

**Comment by hongfaqiu** [comment]
I have already changed the glsl code into glsl300, you can check the [commit history](https://github.com/hongfaqiu/cesium-particle/commits/master) in my repo.
